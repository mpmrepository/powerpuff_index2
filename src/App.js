import React, { useEffect } from "react";
import "./App.scss";
import HomeComponent from "./pages/home/home.container";
import { Switch, Route, NavLink } from "react-router-dom";
import EpisodeComponent from "./pages/episode/episode.component";
import ErrorBoundary from "./ErrorBoundary";
import EpisodesList from "./modules/episode.list";

const App = ({
  onFetchShow,
  onFetchEpisodes,
  showInfo,
  episodesPending,
  episodeList,
  showsPending,
}) => {
  useEffect(() => {
    onFetchShow();
  }, [onFetchShow]);

  useEffect(() => {
    onFetchEpisodes();
  }, [onFetchEpisodes]);

  const { id, image, name } = showInfo;

  const summary = showInfo.summary
    ? showInfo.summary.replace(/(<([^>]+)>)/gi, "")
    : "content loading";

  const pageImage = image
    ? image.original
    : "https://via.placeholder.com/800x1000/ddd/000.png/?text=Image+unavailable!";

  return showsPending ? (
    <div className="App">
      <p>loading</p>
    </div>
  ) : (
    <div className="App">
      <div
        className="cover_image"
        style={{ backgroundImage: `url(${pageImage})` }}
      ></div>
      <div className="app_wrapper">
        <div className="app_content">
          <nav className="pb16">
            <NavLink to={`/`} exact>
              Home
            </NavLink>
          </nav>
          <Switch>
            <Route path={`/:epId`}>
              <ErrorBoundary>
                <EpisodeComponent
                  episodesPending={episodesPending}
                  episodes={episodeList}
                />
              </ErrorBoundary>
            </Route>
            <Route path="/" exact>
              <HomeComponent
                summary={summary}
                id={id}
                name={name}
                showsPending={showsPending}
              />
            </Route>
          </Switch>
          <div className="show_subtitle pb4">
            <h3>Episode list:</h3>
          </div>
          <div className="show_ep_list">
            <EpisodesList episodes={episodeList} />
          </div>
        </div>
      </div>
    </div>
  );
};

export default App;
