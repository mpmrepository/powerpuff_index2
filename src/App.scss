@import './resets.css';

@mixin mobile {
  @media only screen and (max-width:1024px) {
    @content;
  }
}

@font-face {
  font-family: BSTYRENE;
  src: url(https://common.static.burberry.com/fonts/v5/BBY-Styrene/BurberryStyreneB-Regular.woff) format("woff")
}

@font-face {
  font-family: BSTYRENE;
  font-weight: 600;
  src: url(https://common.static.burberry.com/fonts/v5/BBY-Styrene/BurberryStyreneB-Medium.woff) format("woff")
}

@font-face {
  font-family: BSTYRENE;
  font-weight: 700;
  src: url(https://common.static.burberry.com/fonts/v5/BBY-Styrene/BurberryStyreneB-Bold.woff) format("woff")
}

body {
  font-family: BSTYRENE;
  line-height: 1.3;
}

* {
  box-sizing: border-box;
}

h1 {
  font-size: 28px;
}

h2 {
  font-size: 24px;
}

h3 {
  font-size: 14px;
}

h1,
h2 {
  font-weight: 700;
  text-transform: none;
}

h3 {
  font-weight: 700;
  text-transform: none;
}

h4 {
  font-weight: 700;
  text-transform: none;
  font-size: 14px;
}


p {
  font-size: 14px;
  line-height: 22px;
}

.App-header {
  color: black;
  background-color: #f7f7f7;
  padding: 24px 18px;
  margin: 0;
}

.site_header {
  margin: 0;
}

.App-link {
  color: #61dafb;
}


.app_wrapper {
  max-width: 100vw;
  min-height: 100vh;
  display: flex;
  align-items: top;
  justify-content: left;
  box-sizing: border-box;
  align-items: stretch;


}

nav {
  font-size: 12px;
  text-transform: uppercase;
  border-bottom: 2px solid transparent;

  a {
    padding-bottom: 4px;

    &.active {
      border-bottom: 2px solid #000
    }
  }
}

a {
  font-weight: 700;
  color: black;
  text-decoration: none;
  font-size: inherit;



}

.app_content {
  width: 50%;
  order: 1;
  padding: 42px 42px 64px;
  text-align: left;
  box-sizing: border-box;
  overflow: hidden;
  z-index: 10;

  background: #fff;

  @include mobile {
    width: 100%;
    order: 2;

    position: relative;
    float: left;
  }
}

.cover_image {
  width: 50%;
  min-height: 100vh;
  right: 0;
  top: 0;
  order: 1;
  overflow: hidden;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center top;
  position: fixed;
  z-index: 0;

  @include mobile {
    display: none
  }

  img {
    width: 100%;
    height: auto;
  }
}

@for $i from 1 through 20 {
  .pb#{$i} {
    padding-bottom: #{$i * 4}px;
  }
}