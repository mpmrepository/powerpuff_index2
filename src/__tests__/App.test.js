import React from "react";
import { shallow } from "enzyme";
import App from "../App";

describe("App", () => {
  it("should render given props", () => {
    const props = {
      showInfo: {
        id: 6771,
        image:
          "http://static.tvmaze.com/uploads/images/original_untouched/60/151357.jpg",
        name: "Powerpuff Girls",
      },
    };
    const component = shallow(<App {...props} />);
    expect(component).toMatchSnapshot();
  });
});
