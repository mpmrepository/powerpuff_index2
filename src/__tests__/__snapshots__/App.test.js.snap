// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render given props 1`] = `
<div
  className="App"
>
  <div
    className="cover_image"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
  <div
    className="app_wrapper"
  >
    <div
      className="app_content"
    >
      <nav
        className="pb16"
      >
        <NavLink
          exact={true}
          to="/"
        >
          Home
        </NavLink>
      </nav>
      <Switch>
        <Route
          path="/:epId"
        >
          <ErrorBoundary>
            <EpisodeComponent />
          </ErrorBoundary>
        </Route>
        <Route
          exact={true}
          path="/"
        >
          <Connect(HomeComponent)
            id={6771}
            name="Powerpuff Girls"
            summary="content loading"
          />
        </Route>
      </Switch>
      <div
        className="show_subtitle pb4"
      >
        <h3>
          Episode list:
        </h3>
      </div>
      <div
        className="show_ep_list"
      >
        <EpisodesList />
      </div>
    </div>
  </div>
</div>
`;
